khelo/lib/ui/flow/stats/user_stat/user_stat_screen.dart

```diff
@@ -69,360 +44,184 @@ class _UserStatScreenState extends ConsumerState<UserStatScreen>
-                rate.toStringAsFixed(2)))
-      ],
-    );
-  }
+  Widget _fieldingStats(BuildContext context, UserStat? userStat) {
+    if (userStat == null) return Container();
 
-  Widget _fieldingStats(BuildContext context, UserStatViewState state) {
-    final catches = _calculateCatches(state);
-    return Column(
-      crossAxisAlignment: CrossAxisAlignment.start,
-      children: [
-        _sectionTitle(
-            context, context.l10n.my_stat_stats_fielding_statics_title),
-        _runScoredView(context, context.l10n.my_stat_stats_catches_title,
-            catches.toString()),
-        const SizedBox(height: 16),
-        _runOutAndStumpingView(context, state)
-      ],
+    return Container(
+      padding: const EdgeInsets.all(16),
+      margin: const EdgeInsets.only(top: 16),
+      decoration: BoxDecoration(
+          border: Border.all(color: context.colorScheme.outline),
+          borderRadius: BorderRadius.circular(16)),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          _subStatisticView(context,
+              title: context.l10n.my_stat_stats_catches_title,
+              count: userStat.catches.toString()),
+          _subStatisticView(context,
+              title: context.l10n.my_stat_stats_run_out_title,
+              count: userStat.run_out.toString()),
+          _subStatisticView(context,
+              title: context.l10n.my_stat_stats_stumping_title,
+              count: userStat.stumping.toString()),
+        ],
+      ),
     );
   }
 
-  int _calculateCatches(UserStatViewState state) {
-    final catchCounts = state.ballList
-        .where((element) =>
-            element.wicket_taker_id == state.currentUserId &&
-            (element.wicket_type == WicketType.caught ||
-                element.wicket_type == WicketType.caughtBehind ||
-                element.wicket_type == WicketType.caughtAndBowled))
-        .length;
-
-    return catchCounts;
-  }
-
-  Widget _runOutAndStumpingView(BuildContext context, UserStatViewState state) {
-    final (runOut, stumping) = _calculateRunOutAndStumpingCount(state);
+  Widget _sectionTitle(BuildContext context, String title) => Text(title,
+      style: AppTextStyle.header4
+          .copyWith(color: context.colorScheme.textPrimary));
 
-    return Row(
-      children: [
-        Expanded(
-            child: _averageCellView(
-          context,
-          context.l10n.my_stat_stats_run_out_title,
-          runOut.toString(),
-        )),
-        const SizedBox(width: 16),
-        Expanded(
-            child: _averageCellView(
-          context,
-          context.l10n.my_stat_stats_stumping_title,
-          stumping.toString(),
-        )),
-      ],
+  Widget _mainStatisticView(
+    BuildContext context, {
+    required String title,
+    required String count,
+  }) {
+    return Container(
+        width: context.mediaQuerySize.width,
```

Comment:
is there any specific reason, we are giving width like this?

Reply:
yes we need to expand container horizontal.if we not give width then not expand container.

Reply:
how about giving width as **double.infinity**?